samples/bluetooth/broadcast_audio_assistant/src/main.c

```diff
@@ -121,6 +126,84 @@ static bool device_found(struct bt_data *data, void *user_data)
 	}
 }
 
+static bool pa_decode_base(struct bt_data *data, void *user_data)
+{
+	const struct bt_bap_base *base = bt_bap_base_get_base_from_ad(data);
+	uint8_t base_size;
+
+	/* Base is NULL if the data does not contain a valid BASE */
+	if (base == NULL) {
+		return true;
+	}
+
+	base_size = data->data_len - BT_UUID_SIZE_16; /* the BASE comes after the UUID */
+
+	/* Compare BASE and copy if different */
+	if (base_size != received_base_size || memcmp(base, received_base, base_size) != 0) {
+		(void)memcpy(received_base, base, base_size);
+		received_base_size = base_size;
+	}
```

Comment:
Do we need to guard this with a mutex or something, so that we avoid overwriting the BASS while we call `bt_bap_base_foreach_subgroup`?